app/code/Magento/MediaGalleryCatalogUi/Ui/Component/Listing/Columns/Thumbnail.php

```diff
@@ -64,7 +72,7 @@ public function prepareDataSource(array $dataSource)
                 if (isset($item[$fieldName])) {
                     $item[$fieldName . '_src'] = $this->getUrl($item[$fieldName]);
                 } else {
-                    $category = new DataObject($item);
+                    $category = $this->productFactory->create($item);
```

Comment:
I think it should be:
```suggestion
                    $category = $this->productFactory->create(['data' => $item]);
```
Can you please check